### ISSUE TITLE
Travis tripping over AssertionError: Test was too slow

### ISSUE THREAD

**Issue by mih** [datalad/datalad#6412]
This has happened a few times over the past days. Here is one example:

https://app.travis-ci.com/github/datalad/datalad/jobs/558196552
```
======================================================================
4717ERROR: datalad.distribution.tests.test_get.test_source_candidate_subdataset
4718----------------------------------------------------------------------
4719Traceback (most recent call last):
4720  File "/tmp/dl-miniconda-s23vc9zy/lib/python3.9/site-packages/nose/case.py", line 134, in run
4721    self.runTest(result)
4722  File "/tmp/dl-miniconda-s23vc9zy/lib/python3.9/site-packages/nose/case.py", line 152, in runTest
4723    test(result)
4724AssertionError: Test was too slow (took 31.3677s, threshold was 30.0000s)
```

**Comment by yarikoptic** [comment]
Was it happening in the same PR? Then likely it is the one which introduced slow down. I think I never saw this test to exceed its time before

**Comment by yarikoptic** [comment]
FWIW

<details>
<summary>in the past year it seemed always was snappy enough to not hit 30sec mark and had about 5 sec in reserve even in the worst cases</summary> 

```shell
(git)smaug:/mnt/datasets/datalad/ci/logs/2021/12[master]git
$> git grep 'test_source_candidate_subdataset:' | grep travis | sed -e 's,.*: ,,g'| sort -n | nl | tail
   355	23.4473s
   356	23.4852s
   357	23.6095s
   358	23.6279s
   359	23.7064s
   360	23.9314s
   361	24.3871s
   362	24.8944s
   363	25.0311s
   364	25.4870s
1 2796.....................................:Thu 03 Feb 2022 02:47:45 PM EST:.
(git)smaug:/mnt/datasets/datalad/ci/logs/2021/12[master]git
$> cd ../11 
02/  03/  04/  05/  07/  08/  09/  10/  11/  12/  13/  14/  15/  16/  17/  18/  19/  20/  21/  22/  23/  24/  25/  26/  28/  29/  30/
1 2797.....................................:Thu 03 Feb 2022 02:47:53 PM EST:.
(git)smaug:/mnt/datasets/datalad/ci/logs/2021/11[master]git
$> git grep 'test_source_candidate_subdataset:' | grep travis | sed -e 's,.*: ,,g'| sort -n | nl | tail
   199	21.0065s
   200	21.2269s
   201	21.2856s
   202	21.3108s
   203	21.3109s
   204	21.4974s
   205	21.8706s
   206	22.2052s
   207	22.6064s
   208	24.3214s

```
</details>

<details>
<summary>situation changed in January </summary> 

```shell
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/01[master]git
$> git grep 'test_source_candidate_subdataset:' | grep travis | sed -e 's,.*: ,,g'| sort -n | nl | tail
   243	27.8073s
   244	28.3062s
   245	28.3536s
   246	28.4859s
   247	28.8344s
   248	28.9748s
   249	29.6000s
   250	30.2112s
   251	30.5768s
   252	31.2354s
```
</details>

```
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/01[master]git
$> git grep 'test_source_candidate_subdataset:' | grep push/master | tail
27/push/master/01793cc/travis-13332-success/8.txt:[success] 1.85% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 22.3623s
29/push/master/9aa3a8c/travis-13369-success/4.txt:[success] 2.49% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 24.0958s
29/push/master/9aa3a8c/travis-13369-success/7.txt:[success] 2.45% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 25.8856s
29/push/master/9aa3a8c/travis-13369-success/8.txt:[success] 2.27% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 25.8270s
29/push/master/f433f86/travis-13368-failed/8.txt:[error] 2.35% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 30.2112s
29/push/master/f433f86/travis-13368-success/4.txt:[success] 2.47% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 24.5627s
29/push/master/f433f86/travis-13368-success/7.txt:[success] 2.43% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 28.8344s
31/push/master/fd8a3cc/travis-13379-success/4.txt:[success] 2.45% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 23.9239s
31/push/master/fd8a3cc/travis-13379-success/7.txt:[success] 2.49% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 25.9854s
31/push/master/fd8a3cc/travis-13379-success/8.txt:[success] 2.26% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 28.4859s
```
so likely somewhere in 01793cc..f433f86 range... 
since the test seems to be concerned about `ria+` I thought it relates to https://github.com/datalad/datalad/pull/6355

<details>
<summary>but it was always fast in that pr</summary> 

```shell
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/01[master]git
$> git grep 'test_source_candidate_subdataset:' | grep pr/6355 | tail   
19/pr/6355/96fd3e8/travis-13291-success/7.txt:[success] 1.82% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 19.5946s
19/pr/6355/ef0c695/travis-13296-success/3.txt:[success] 1.97% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 17.2688s
19/pr/6355/ef0c695/travis-13296-success/6.txt:[success] 1.94% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 17.5066s
19/pr/6355/ef0c695/travis-13296-success/7.txt:[success] 1.78% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 20.6147s
26/pr/6355/5927b9f/travis-13322-success/4.txt:[success] 2.50% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 22.6064s
26/pr/6355/5927b9f/travis-13322-success/7.txt:[success] 2.55% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 25.5250s
26/pr/6355/5927b9f/travis-13322-success/8.txt:[success] 2.30% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 25.0657s
26/pr/6355/6d88a57/travis-13312-success/3.txt:[success] 1.95% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 19.0808s
26/pr/6355/6d88a57/travis-13312-success/6.txt:[success] 2.08% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 20.0240s
26/pr/6355/6d88a57/travis-13312-success/7.txt:[success] 1.83% datalad.distribution.tests.test_get.test_source_candidate_subdataset: 19.3359s

```
</details>

so I give up for now.  But note that looking at the test which doesn't even involve remote connections, 30 sec for the seems to be quite a bit. May be @bpoldrack could shine more light on what takes most of the time, and may be things could be improved.  Otherwise we just need to announce the test `@slow`.

**Comment by bpoldrack** [comment]
> since the test seems to be concerned about ria+ I thought it relates to #6355
> 
> but it was always fast in that pr

Yeah, also implausible, since that PR was only concerned with HTTP and the test uses local access only.

>  May be @bpoldrack could shine more light on what takes most of the time

At least it isn't obvious to me either. It does involve the creation of 13 (very small) repositories in total, though (either via `datalad create` or via `git init` from within `datalad create-sibling-ria` or via `git clone`). So, I guess `create` and/or `clone` may have become slower.
In short: Don't see an immediate solution. This will require some serious profiling.

**Comment by yarikoptic** [comment]
FWIW ran for those two commits

01793cc

```
1: python -m nose -s -v datalad/distribution/tests/test_get.py:test_source_candidate_subdataset
            Mean        Std.Dev.    Min         Median      Max
real        31.350      2.827       29.518      30.012      36.979      
user        13.256      1.101       12.386      12.766      15.405      
sys         10.332      1.026       8.377       10.651      11.424 
```

f433f86 
```
1: python -m nose -s -v datalad/distribution/tests/test_get.py:test_source_candidate_subdataset
            Mean        Std.Dev.    Min         Median      Max
real        29.757      0.725       28.885      29.424      30.878      
user        12.607      0.454       11.826      12.620      13.245      
sys         10.121      1.108       8.070       10.755      10.999 
```

and then went even more back to 1304efa0108646430d1787a5841144abb5ec82ff

```
            Mean        Std.Dev.    Min         Median      Max
real        29.686      1.266       27.618      30.310      30.871      
user        12.777      0.341       12.300      13.014      13.119      
sys         10.061      0.753       8.884       9.889       10.890  
```

that is with

- dependencies: cmd:git=2.35.0 cmd:annex=10.20220127-gf616630 cmd:bundled-git=UNKNOWN cmd:system-git=2.35.0 cmd:system-ssh=8.8p1 cmd:7z=16.02 annexremote=1.5.0 platformdirs=2.4.1 boto=2.49.0 exifread=2.3.2 humanize=3.14.0 iso8601=1.0.2 keyring=23.4.0 keyrings.alt=UNKNOWN msgpack=1.0.3 mutagen=1.45.1 requests=2.27.1

so may be indeed it is the effect of changed externals -- unfortunately we are having power shutdown, so smaug is out of reach, can't verify if the same git annex build was used.

so , may be indeed let's just announce `@slow` and be done with it for now?

**Comment by bpoldrack** [comment]
> FWIW ran for those two commits
> 01793cc
> ...

Just a quick trial, but on my laptop:
```
...
Ran 1 test in 19.952s

OK
python -m nose -s -v   9.88s user 9.01s system 92% cpu 20.514 total
```
That's quite a difference on the same commit. However, I noticed: `cmd:annex=8.20211123`, which kinda fits your digging into the logs that would suggest it became slower during December+January.

> so , may be indeed let's just announce @slow and be done with it for now?

I am kinda fine with that. But we need to make sure, we gonna look into it in more detail. Otherwise it becomes an effective `skip`.
